src/components/application_manager/src/resumption/resume_ctrl_impl.cc

```diff
@@ -219,6 +230,89 @@ bool ResumeCtrlImpl::RestoreAppHMIState(ApplicationSharedPtr application) {
   return result;
 }
 
+void ResumeCtrlImpl::RestoreWidgetsHMIState(
+    const smart_objects::SmartObject& response_message) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const auto correlation_id =
+      response_message[strings::params][strings::correlation_id].asInt();
+  const auto& request = requests_msg_.find(correlation_id);
+  if (requests_msg_.end() == request) {
+    LOG4CXX_ERROR(logger_,
+                  "Request UI_CreateWindow for correlation id: "
+                      << correlation_id << " not found");
+    return;
+  }
+
+  const auto& msg_params = (*request->second)[strings::msg_params];
+  const auto hmi_app_id = msg_params[strings::app_id].asInt();
+  auto application = application_manager_.application_by_hmi_app(hmi_app_id);
+  if (!application) {
+    LOG4CXX_ERROR(logger_,
+                  "Application is not registered by hmi id: " << hmi_app_id);
+    requests_msg_.erase(request);
+    return;
+  }
+  const WindowID window_id = msg_params[strings::window_id].asInt();
```

Comment:
I see a lot of mixing this parameter as an Int and as a UInt.

Reply:
right, at the beginning it was unsigned type which was changed to signed one during internal review. changed all uses to signed in 4a8ee34c2fb59dd4cd8789df38f6642c95e19bb9